### PR TITLE
Add gh actions

### PR DESCRIPTION
**Pull request by ffromani**
automate the test builds and the unit tests runs for basic sanity checks;
we can add a lane to automatically create container images later.

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Set up golang
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.16
+
+    - name: Build binaries
```
Comment (swatisehgal) [inline .github/workflows/go.yml:26]: The PR looks good to me but what do you think about adding a format check here?

In a separate PR, maybe we could add workflows for e2e tests as well. Not a blocker in anyway for this PR!

Comment (ffromani) [reply .github/workflows/go.yml:26]: format like in `go fmt`? sure thing

Comment (ffromani) [reply .github/workflows/go.yml:26]: e2e + artifacts creation + push to container registry will toatally come in followup PRs

**Pull request merged**